index.js

```diff
@@ -1,73 +1,90 @@
-const { Client, Collection, GatewayIntentBits } = require('discord.js');
-const { token, guildId } = require('./config.json');
-
-const fs = require('fs')
-const path = require("path")
+const { Client, Collection, GatewayIntentBits, Events } = require('discord.js');
+const { token } = require('./config.json');
+const fs = require('fs');
+const path = require('path');
 // Create a new client instance
-const client = new Client({ intents: [GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages, GatewayIntentBits.DirectMessages, GatewayIntentBits.GuildVoiceStates, GatewayIntentBits.GuildMembers] });
+const client = new Client({
+  intents: [
+    GatewayIntentBits.Guilds,
+    GatewayIntentBits.GuildMessages,
+    GatewayIntentBits.DirectMessages,
+    GatewayIntentBits.GuildVoiceStates,
+    GatewayIntentBits.GuildMembers,
+  ],
+});
 
 // Register Commands
 client.commands = new Collection();
-const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
+const commandFiles = fs.readdirSync('./commands').filter((file) => file.endsWith('.js'));
 
 for (const file of commandFiles) {
-	const command = require(`./commands/${file}`);
-	client.commands.set(command.data.name, command);
+  const command = require(`./commands/${file}`);
+  client.commands.set(command.data.name, command);
 }
 
 
 
 client.once('ready', () => {
-	console.log('Ready!');
+  console.log('Ready!');
 });
 
-client.on('interactionCreate', async interaction => {
-	if (!interaction.isCommand()) return;
+client.on(Events.InteractionCreate, async (interaction) => {
+  try {
+    if (interaction.isButton()) {
+      const buttonUser = interaction.user.id;
+      const commandUser = interaction.message.interaction.user.id;
+      if (buttonUser === commandUser) {
+        if (interaction.message.content.includes('Blackjack')) {
+          const command = client.commands.get('blackjack');
+          if (!command) return;
 
-	const command = client.commands.get(interaction.commandName);
-	if (!command) return;
-
-	try {
-		await command.execute(interaction);
-	} catch (error) {
-		console.log(error)
-	}
-});
+          await command.execute(interaction, interaction.customId);

```

Comment:
so you are using the interaction's custom id to store this?

Reply:
Is this customid unique for each user if multiple people are playing at the same time?

Reply:
The customId will be unique to each interaction. So if multiple players are playing, itll create a different interaction and therefore should unique for each player. However, I didn't get to test this with multiple people so I can be very wrong.

Reply:
We can try and test this in my test server when you got some time!